### PR TITLE
config: change default prometheus scrape interval

### PR DESCRIPTION
**Pull request by mcalman**
Addresses #963

This makes "30s" the new Kepler default scraping interval for Prometheus. It also adds a new manifest build option (`"OPTS="PROMETHEUS_DEPLOY HIGH_GRANULARITY" make build-manifest`) to revert to the original "3s" scrape interval.

Users can also specify any scraping interval they would like by modifying PROMETHEUS_SCRAPE_INTERVAL in the ConfigMap before building the manifests.

If accepted, I will add the new deployment option to the docs [here](https://github.com/sustainable-computing-io/kepler-doc/blob/aa3a0850db39918bba5d721b6ae27faad5a03efb/docs/installation/kepler.md?plain=1#L67).

### CODE REVIEW

File: manifests/config/base/kustomization.yaml [hunk 1]
```diff
@@ -25,4 +25,6 @@ patchesStrategicMerge: []
 # add this line to set model-server endpoint to kepler
 # - ./patch/patch-model-server-kepler-config.yaml
 # add this line to change log level for debugging
-# - ./patch/patch-debug.yaml
\ No newline at end of file
+# - ./patch/patch-debug.yaml
+# add this line to get high metric granularity (3s interval)
+# - ./patch/patch-high-granularity.yaml
```
Comment (rootfs) [inline manifests/config/base/kustomization.yaml:30]: is this file coming next?

Comment (mcalman) [reply manifests/config/base/kustomization.yaml:30]: @rootfs good catch. I have added the file

**Review by rootfs — APPROVED** [review]

**Comment by sunya-ch** [comment]
> @sunya-ch can we increase the Prometheus scrapping interval?

Yes. I think it is fine. Here are results of stressng stressing on CPU with 7 different stress load on VM using estimator sidecar with non-linear power model (with PR https://github.com/sustainable-computing-io/kepler/pull/976).

Screenshot 2023-10-10 at 19 16 45

Screenshot 2023-10-10 at 19 17 36

**Comment by marceloamaral** [comment]
@sunya-ch how long the workload was running?

The impact in the accuracy looks significant.... :/

**Comment by sthaha** [comment]
@mcalman, thank you for the patch. We were thinking of providing a similar feature in Kepler operator to control the `fidelity | resolution | granularity` 

@marceloamaral, @rootfs, In this regard I have had a discussion offline with @simonpasquier (who is a prometheus maintainer) about the scrape interval. It seems having too low (3s) scrape interval can lead to failed scapes when the amount of metrics produced is too large. 

IIRC, the recommendation was to keep the scrape interval between 15s-2m and have prometheusrules (if any) to be evaluated at the same or greater interval

@simonpasquier, could you please shed more light on this?

**Review by marceloamaral — CHANGES_REQUESTED** [review]
@mcalman
I think the default Prometheus scrapping interval is 15s in Vanila Kubernetes and 30s in OpenShift.
Can you update it to 15s and maybe patch to 30s for OpenShift deployment?

**Comment by mcalman** [comment]
> @mcalman I think the default Prometheus scrapping interval is 15s in Vanila Kubernetes and 30s in OpenShift. Can you update it to 15s and maybe patch to 30s for OpenShift deployment?

@marceloamaral It seems different versions of Prometheus also have different defaults. For example, the version deployed in the [kube-prometheus stack](https://github.com/prometheus-community/helm-charts/blob/29916435a6c1b157b5372a3a398808a87dca9517/charts/kube-prometheus-stack/values.yaml#L2878C6-L2878C6) scrapes at 30s intervals by default. If we'd like to always use the Prometheus default value we can also remove the `interval` field from [prometheus_service_monitor.yaml](https://github.com/sustainable-computing-io/kepler/pull/970/files#diff-f3746997fef4bd7ed174d9241e8b4b47127f55f5c0487ed3901e9831f1f548ea) which will cause Prometheus's global scrape interval to be used.

Should we only specify a custom Prometheus scrape interval for Kepler when the `HIGH_GRANULARITY` option is set and let the global scrape interval be used otherwise?

**Comment by sthaha** [comment]
The default scrapeInterval defined by Prometheus CRD is 30s - https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1/prometheus_types.go#L217

**Review by marceloamaral — APPROVED** [review]
/lgtm

Great, let's set it to 30s then!

**Pull request merged**